### PR TITLE
[Bug][Vectorized] Support array function in where pre in volap_scan_node

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #10388

## Problem Summary:

95B8EC66470E8334E5CB50FE0EF3F58F

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by cambyzju** [comment]
LGTM

**Comment by eldenmoon** [comment]
LGTM

File: be/src/exprs/array_functions.h [hunk 1]
```diff
@@ -32,4 +32,6 @@ class ArrayFunctions {
 
     static CollectionVal array(FunctionContext* context, int num_children, const StringVal* values);
 };
+
+doris_udf::AnyVal array_fake_fucntion(FunctionContext* context);
```
Comment (yinzhijian) [inline be/src/exprs/array_functions.h:36]: array_fake_function

File: fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java [hunk 2]
```diff
@@ -1263,6 +1263,11 @@ public void addScalarAndVectorizedBuiltin(String fnName, String symbol, boolean
                                               boolean varArgs, Type ... args) {
         ArrayList<Type> argsType = new ArrayList<Type>();
         for (Type type : args) {
+            // only to prevent olap scan node use array expr to find a fake symbol
+            // TODO: delete the code after we remove origin exec engine
+            if (type.isArrayType()) {
+                symbol = "_ZN5doris19array_fake_fucntionEPN9doris_udf15FunctionContextE";
```
Comment (yinzhijian) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java:1269]: array_fake_function

**Review by xy720 — DISMISSED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**